pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -509,3 +529,595 @@ def reconstructing_fractional_octave_bands(
+           'Electroacoustics: Octave-Band and Fractional-Octave-Band
+           Filters.(IEC61260)', Part 1, Part 1, (2014),
+           ISBN: 978-2-8322-1417-6
+    """
+
+    def __init__(
+            self,
+            num_fractions=1,
+            frequency_range=(20.0, 20e3),
+            order=14,
+            sampling_rate=44100):
+
+        self._num_fractions = num_fractions
+        self._frequency_range = frequency_range
+        self._order = order
+        self._sampling_rate = sampling_rate
+
+        self._nominal_frequencies,\
+            self._center_frequencies,\
+            self._cutoff_frequencies = fractional_octave_frequencies(
+                self._num_fractions, self._frequency_range, True)
+
+        coefficients = self._get_coefficients(
+                sampling_rate=self._sampling_rate,
+                num_fractions=self._num_fractions,
+                frequency_range=self._frequency_range,
+                order=self._order)
+
+        # initialize superclass
+        super().__init__(coefficients, self._sampling_rate)
+        self.comment = (
+            f"Second order section 1/{num_fractions} fractional octave band"
+            "filter of order {order}")
+
+    def __repr__(self):
+        """Nice string representation of class instances"""
+        return (f"Energy-preserving fractional "
+                f"octave filter bank with {self.n_bands} "
+                f"bands between {self.frequency_range[0]} and "
+                f"{self.frequency_range[1]} Hz with "
+                f"{self.sampling_rate} Hz sampling rate and "
+                f"{self.num_fractions} per octave.")
+
+    def __eq__(self, other):
+        """Check for equality of two objects."""
+        return not DeepDiff(self.__dict__, other.__dict__)
+
+    @property
+    def num_fractions(self):
+        """Get the number of fractions per octave"""
+        return self._num_fractions
+
+    @property
+    def frequency_range(self):
+        """Get the frequency range of the filter bank in Hz"""
+        return self._frequency_range
+
+    @property
+    def nominal_frequencies(self):
+        """Get the IEC center frequencies of the (fractional)
+         octave bands in Hz"""
+        return self._nominal_frequencies
+
+    @property
+    def center_frequencies(self):
+        """Get the exact center frequencies of the (fractional)
+        octave bands in Hz"""
+        return self._center_frequencies
+
+    @property
+    def cutoff_frequencies(self):
+        """Get the cutoff frequencies of the (fractional)
+        octave bands in Hz"""
+        return self._cutoff_frequencies
+
+    @property
+    def n_bands(self):
+        """Get the number of bands in the filter bank"""
+        return len(self._center_frequencies)
+
+    def _get_coefficients(
+            self,
+            sampling_rate,
+            num_fractions,
+            frequency_range,
+            order):
```

Comment:
```suggestion
    @property
    def coefficients(self):
```
This can be converted into a property, overloading the existing coefficients property.
All input variables to this method are part of the class and hence can be accessed inside the class without passing them as input parameters.

Reply:
Doing that causes an `AttributeError` in `pyfar/classes/filter.py:139`. It seems that me overloading the `coefficients` property disables the `@coefficients.setter` in the base class, causing the property to be impossible to set. If I then add this setter to the `FractionalOctaveBands` class as well, i don't know where to call it, as the property is not recognized in the constructor. I'm essentially trying to recreate `pyfar/classes/filter.py:139`.

Reply:
There's a tricky issue with that. Simply overloading an returning the coefficients causes an `AttributeError: can't set attribute 'coefficients'`. Taking a look at lines `171`-`174` of `filter.py` shows that I need to specify the property's `setter` explicitly. However, this `setter` gets called on line `139` if `coefficients` is passed along to the filter's constructor, which does not happen in my case, as the constructor of `FractionalOctaveBands` does not take any coefficients. These get created from more basic arguments. In essence, i think that the parent class is forcing me to go with a `setter` approach, even though I don't have/require a `setter`. A simple 
```python
@property
def coefficients(self):
    # calc
    return sos_result
```
should be enough, but again, this throws an error.

Btw, giving the property a different name fixes the issue, so it's definetely the base class messing with the implementation.

Reply:
Not sure if I fully understand the issue. Can you commit or give a minimal example that raises the error. It is possible to explicitly override properties, would this help: https://stackoverflow.com/questions/7019643/overriding-properties-in-python